src/include/mallocMC/allocator.hpp

```diff
@@ -111,90 +130,120 @@ namespace mallocMC
         using AlignmentPolicy = T_AlignmentPolicy;
         using HeapInfoVector = std::vector<HeapInfo>;
         using DevAllocator = DeviceAllocator<
+            AlpakaAcc,
             CreationPolicy,
             DistributionPolicy,
             OOMPolicy,
             AlignmentPolicy>;
-        using AllocatorHandle = AllocatorHandleImpl<Allocator>;
+        using AllocatorHandle = AllocatorHandleImpl<AlpakaAcc, Allocator>;
 
     private:
-        AllocatorHandle allocatorHandle;
+        std::function<void()>
+            freeLastAlloc; // calles free(dev), with the dev used during alloc
+        decltype(alpaka::mem::buf::alloc<
+                 DevAllocator,
+                 alpaka::idx::traits::IdxType<AlpakaAcc>::type>(
+            std::declval<const alpaka::dev::Dev<AlpakaAcc>>(),
+            std::declval<
+                const alpaka::idx::traits::IdxType<AlpakaAcc>::type>()))
```

Comment:
Yes, these long types are an unfortunate side of alpaka. We often try to manage it by first creating a helper `using` type aliases with readable name, to be used later in more complex constructions. Something along the lines of
```cpp
using Dim = alpaka::dim::Dim<AlpakaAcc>;
using Idx = alpaka::idx::Idx<AlpakaAcc>;
using Dev = alpaka::dev::Dev<AlpakaAcc>;
using Buf = alpaka::mem::buf::Buf<Dev, YourDataType, Dim, Idx>;
```

Reply:
> Because that decltype is just too big.

What do you mean by `too big`?

> what is the type of a buffer in alpaka?

Do you mean how to create a buffer with alpaka? https://github.com/alpaka-group/alpaka/blob/a5a8277cded3e20612c11282857ad2f007a113f3/example/vectorAdd/src/vectorAdd.cpp#L137-L138

Reply:
@psychocoderHPC: I mean this declaration I have put into Allocator:
```
decltype(alpaka::mem::buf::alloc<
                 DevAllocator,
                 alpaka::idx::traits::IdxType<AlpakaAcc>::type>(
            std::declval<const alpaka::dev::Dev<AlpakaAcc>>(),
            std::declval<
                const alpaka::idx::traits::IdxType<AlpakaAcc>::type>()))
            devAllocatorStorage;
```

@sbastrakov: I will try your using declaration!

Reply:
Yes, 7 line `decltype` qualifies as too big, even for alpaka standards xd.

Reply:
improved